Shared/Samples/Geocode offline/GeocodeOfflineView.Model.swift

```diff
@@ -0,0 +1,170 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension GeocodeOfflineView {
+    /// The view model for the sample.
+    @MainActor
+    class Model: ObservableObject {
+        // MARK: Properties
+        
+        /// A map with a tiled layer of the streets in San Diego, CA, USA.
+        let map = {
+            // Create a tiled layer using the local tile package.
+            let tileCache = TileCache(fileURL: .streetMap)
+            let tiledLayer = ArcGISTiledLayer(tileCache: tileCache)
+            
+            // Create a map with the tiled layer as base layer.
+            return Map(basemap: Basemap(baseLayer: tiledLayer))
+        }()
+        
+        /// The graphics overlay for the marker graphic.
+        let graphicsOverlay = GraphicsOverlay()
+        
+        /// The red map marker graphic used to indicate a given location on the map.
+        private let markerGraphic = {
+            // Create a symbol using the image from the project assets.
+            guard let markerImage = UIImage(named: "RedMarker") else { return Graphic() }
+            let markerSymbol = PictureMarkerSymbol(image: markerImage)
+            
+            // Change the symbol's offsets, so it aligns properly to a given point.
+            markerSymbol.leaderOffsetY = markerImage.size.height / 2
+            markerSymbol.offsetY = markerImage.size.height / 2
+            
+            // Create a graphic with the symbol.
+            return Graphic(symbol: markerSymbol)
+        }()
+        
+        /// The locator task used to preform the geocode operations, loaded from a local file.
+        private let locatorTask = LocatorTask(name: "SanDiego_StreetAddress", bundle: .main)
+        
+        /// The placement of the callout on the map.
+        @Published var calloutPlacement: CalloutPlacement?
+        
+        /// A Boolean value indicating whether the callout placement should be offsetted, e.g. when the map magnifier is showing.
+        @Published var calloutShouldOffset = false
+        
+        /// The text shown in the callout.
+        @Published private(set) var calloutText: String = ""
+        
+        init() {
+            graphicsOverlay.addGraphic(markerGraphic)
+        }
+        
+        // MARK: Methods
+        
+        /// Geocodes a given address and adds a marker with the corresponding address at the result's location.
+        /// - Parameter address: The given text address to geocode.
+        /// - Returns: The extent of the result's display location.
+        func geocodeSearch(address: String) async -> Envelope? {
+            guard let locatorTask else { return nil }
+            
+            // Create geocode parameters.
+            let geocodeParameters = GeocodeParameters()
```

Comment:
I'm in favor of keeping the GeocodeParameters construction as local variables, even if they don't change between searches and can be held as properties. Reasons: construction is rather low cost, and it shows a clearer workflow that a geocode/reverseGeocode requires parameters to be specifically configured. When a user copy this snippets, they can easily modify it to suit their usecase.